src/hooks/screen/useChatScreenHook.ts

```diff
@@ -182,10 +221,17 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
           null
         );
       }
-
-      await currentChatSchema.updateChatSentStatus(localDb, response);
-      refresh('chat');
-      refresh('channelList');
+      queue.addPriorityJob({
+        operationLabel: DatabaseOperationLabel.ChatScreen_UpdateChatSentStatus,
+        id: `${selectedChannel?.id}-${randomId}`,
+        priority: QueueJobPriority.HIGH,
+        forceAddToQueue: true,
+        task: async () => {
+          await currentChatSchema?.updateChatSentStatus(localDb, response);
+          initChatData();
+          refresh('channelList');
+        }
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of the `updateChatSentStatus` function is a significant improvement over the previous one. It uses the queue to manage database operations, which can enhance performance and maintainability. However, there's no error handling for the tasks added to the queue. If an error occurs during the execution of these tasks, it might lead to unexpected behavior or crashes. Consider adding error handling within each task in the queue.
```diff
      queue.addPriorityJob({
        operationLabel: DatabaseOperationLabel.ChatScreen_UpdateChatSentStatus,
        id: `${selectedChannel?.id}-${randomId}`,
        priority: QueueJobPriority.HIGH,
        forceAddToQueue: true,
        task: async () => {
          try {
            await currentChatSchema?.updateChatSentStatus(localDb, response);
            initChatData();
            refresh('channelList');
          } catch (e) {
            console.error('Error while updating chat sent status:', e);
          }
        }
      });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->